cis194/week8/zhansongl/Party.hs

```diff
@@ -0,0 +1,46 @@
+{-# OPTIONS_GHC -Wall #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Party where
+
+import Employee
+import Data.Tree (Tree, rootLabel, subForest)
+
+glCons :: Employee -> GuestList -> GuestList
+glCons e (GL l f) = GL (e:l) (f + empFun e)
+
+instance Semigroup GuestList where
+  (<>) (GL l1 f1) (GL l2 f2) = GL (l1 ++ l2) (f1 + f2)
+
+instance Monoid GuestList where
+  mempty = GL [] 0
+
+getFun :: GuestList -> Fun
+getFun (GL _ f) = f
+
+moreFun :: GuestList -> GuestList -> GuestList
+moreFun gl1 gl2 = case compare (getFun gl1) (getFun gl2) of
+                    GT -> gl1
+                    _  -> gl2
+
+treeFold :: (a -> b -> b) -> b -> Tree a -> b
```

Comment:
I initially wrote this signature until I saw https://hackage.haskell.org/package/containers-0.6.2.1/docs/Data-Tree.html#v:foldTree